--- libpng-1.0.59/contrib/pngminus/makefile.std.orig	2012-03-29 07:00:13.000000000 +0200
+++ libpng-1.0.59/contrib/pngminus/makefile.std	2012-04-25 19:26:45.070508661 +0200
@@ -19,11 +19,11 @@
 #ZINC = -I$(ZPATH)/include
 #ZLIB = -L$(ZPATH)/lib -lz
 #ZLIBS = $(ZPATH)/lib/libz.a
-ZINC = -I../../../zlib
-ZLIB = -L../../../zlib -lz
-ZLIBS = ../../../zlib/libz.a
+ZINC = 
+ZLIB = -lz
+ZLIBS = -lz
 
-CFLAGS=$(PNGINC) $(ZINC)
+CFLAGS=$(OPT_FLAGS) $(PNGINC) $(ZINC)
 LDLIBS=$(PNGLIB) $(ZLIB)
 LDLIBSS=$(PNGLIBS) $(ZLIBS)
 C=.c
